tests/test_sharing_client.py

```diff
@@ -0,0 +1,229 @@
+                                                                                  expected_scope, site_id, 1,
+                                                99999, 86400)
+                self._client.refresh_keys()
+                with self.assertRaises(EncryptionError) as context:
+                    self._client.decrypt_token_into_raw_uid(token)
+                self.assertEqual('invalid token lifetime', str(context.exception))
+
+    def test_token_generated_in_the_future_within_allowed_clock_skew(self, mock_refresh_sharing_keys):  # TokenGeneratedInTheFutureWithinAllowedClockSkew
+        created_at_future = dt.datetime.now(tz=timezone.utc) + dt.timedelta(minutes=29)  #max allowed clock skew is 30m
+        for expected_scope, expected_version in test_cases_all_scopes_all_versions:
+            with self.subTest(expected_scope=expected_scope, expected_version=expected_version):
+                token = generate_uid_token(expected_scope, expected_version, expires_at=created_at_future)
+                mock_refresh_sharing_keys.return_value = EncryptionKeysCollection([master_key, site_key],
+                                                                                  expected_scope, site_id, 1,
+                                                99999, 86400)
+                self._client.refresh_keys()
+                result = self._client.decrypt_token_into_raw_uid(token)
+                self.assertIsNotNone(result)
+                self.assertEqual(result.identity_scope, expected_scope)
+                self.assertEqual(result.advertising_token_version, expected_version)
+
+    def test_phone_uids(self, mock_refresh_sharing_keys):  # PhoneTest
+        for expected_scope, expected_version in test_cases_all_scopes_v3_v4_versions:
+            with self.subTest(expected_scope=expected_scope, expected_version=expected_version):
+                mock_refresh_sharing_keys.return_value = EncryptionKeysCollection([master_key, site_key],
+                                                                                  expected_scope, site_id, 1,
+                                                                                  99999, 86400)
+                self._client.refresh_keys()
+                token = generate_uid_token(expected_scope, expected_version, phone_uid)
+                self.assertEqual(IdentityType.Phone, get_identity_type(token))
+                result = self._client.decrypt_token_into_raw_uid(token)
+                self.assertIsNotNone(result)
+                self.assertEqual(result.uid2, phone_uid)
+                self.assertEqual(result.identity_scope, expected_scope)
+                self.assertEqual(result.advertising_token_version, expected_version)
+
+    # Tests below taken from test_client.py related to Sharing
+
+    def test_sharing_client_produces_uid2_token(self, mock_refresh_keys_util):  #ClientProducesTokenWithCorrectPrefix
```

Comment:
Better not to rename tests if we can avoid it. This tests the human-readable prefix specifically, not the binary bit that determines UID2 vs EUID